### PR TITLE
sql/catalog/lease: wait until the descriptor is visible at now

### PR DESCRIPTION
**Pull request by ajwerner**
In MR serverless, we're making the lease table a global table. We want to make sure we wait until new leases will find the newest version, lest we drop leases at the current version and then have to re-acquire, and, worse, hit the uncertainty interval.

Epic: none

Release note: None

### CODE REVIEW

File: pkg/sql/catalog/lease/lease.go [hunk 1]
```diff
@@ -1130,28 +1130,49 @@ func (m *Manager) RefreshLeases(ctx context.Context, s *stop.Stopper, db *kv.DB)
...
-					log.Warningf(ctx, "error purging leases for descriptor %d(%s): %s",
-						id, name, err)
+					desc.GetID(), desc.GetVersion(), dropped)
+				purge := func(ctx context.Context) {
+					if err := purgeOldVersions(ctx, db, desc.GetID(), dropped, desc.GetVersion(), m); err != nil {
+						log.Warningf(ctx, "error purging leases for descriptor %d(%s): %s",
+							desc.GetID(), desc.GetName(), err)
+					}
+				}
+				// New descriptors may appear in the future if the descriptor table is
```
Comment (chengxiong-ruan) [inline pkg/sql/catalog/lease/lease.go:1150]: Why the table's modification time can be in the future? It's because the clock difference between regions?

**Review by chengxiong-ruan — APPROVED** [review]
LGTM!

**Comment by ajwerner** [comment]
TFTR!

bors r+

**Comment by knz** [comment]
should this be backported?

**Pull request merged**